amazon_cloudwatch_agent.sh

```diff
@@ -17,4 +26,31 @@ make -f /usr/share/selinux/devel/Makefile amazon_cloudwatch_agent.pp || exit
 /sbin/restorecon -R -v /opt/aws/amazon-cloudwatch-agent || true
 /sbin/restorecon -v /etc/systemd/system/amazon-cloudwatch-agent.service || true
 
-echo "Policy loaded. You may need to restart the CloudWatch agent: systemctl restart amazon-cloudwatch-agent"
\ No newline at end of file
+echo "Adding environment variable to systemd service"
+
+# Ensure the directory exists
+sudo mkdir -p /etc/systemd/system/amazon-cloudwatch-agent.service.d
+
+# Create the override file with the Environment variable
+cat <<EOF | sudo tee /etc/systemd/system/amazon-cloudwatch-agent.service.d/override.conf
+[Service]
+Environment="RUN_WITH_SELINUX=True"
+EOF
+
+# Reload systemd
+sudo systemctl daemon-reload
+
+# Check if auto-approve is enabled, otherwise ask the user
+if [ "$AUTO_APPROVE" = true ]; then
+    echo "Auto-approve enabled. Restarting CloudWatch Agent..."
+    sudo systemctl restart amazon-cloudwatch-agent
+    echo "CloudWatch Agent has been restarted."
```

Comment:
nit: can you just exit here instead of a if else statement. I think no nest is usually easier to maintain